tests/RestServer/LanguageFeatureTest.cpp

```diff
@@ -127,6 +126,30 @@ void checkLanguageFile(const arangodb::ArangodServer& server,
     ASSERT_TRUE(false);
   }
 }
+
+constexpr std::string_view kGerman = "de";
+constexpr std::string_view kRussian = "ru";
+
+std::string_view getNonSysLang() {
+  auto before = std::string_view(setlocale(LC_ALL, NULL)).substr(0, 2);
+  setlocale(LC_ALL, "");
+  const std::string_view currSysLang =
+      std::string_view(setlocale(LC_ALL, NULL)).substr(0, 2);
+
+  auto sg = arangodb::scopeGuard([&]() noexcept {
+    setlocale(LC_ALL, before.data());
```

Comment:
This won't work as expected.
Although `before` now has a length of (up to) 2 bytes, using only its data pointer will point to a string of that is not necessarily just 2 bytes long.

Try running this program:
```
#include <string_view>
#include <iostream>

int main() {
  char const* loca = "de_DE";
  auto before = std::string_view(loca).substr(0, 2);

  std::cout << "before: "        << before << "\n";
  std::cout << "before.data(): " << before.data() << "\n";
}
```
This prints
```
$ g++ -std=c++20 sv.cpp && ./a.out 
before: de
before.data(): de_DE
```
Whenever you need to guarantee that the strings are null-terminated, it is best to use std::string and not std::string_view.
And you need the null-termination because the pointer is passed into a C function.